moshi/src/main/java/com/squareup/moshi/internal/Util.java

```diff
@@ -69,6 +70,11 @@
     DEFAULT_CONSTRUCTOR_MARKER = defaultConstructorMarker;
   }
 
+  @Keep /* Kept to avoid optimizing out reflection. */
```

Comment:
You can use our existing rules file to keep this rather than using the dependency.